### PR TITLE
Direct Write in ERROR_IF_EXISTS mode works

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryCatalog.java [hunk 1]
```diff
@@ -114,6 +121,7 @@ public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchT
   @Override
   public boolean dropTable(Identifier ident) {
     // TODO: add implementation
+    bigQueryClient.deleteTable(((BigQueryIdentifier) ident).getTableId());
```
Comment (VinayLondhe14) [inline spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryCatalog.java:124]: Shouldn't we return true below?

Comment (davidrabinowitz) [reply spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryCatalog.java]: fixed

**Review by VinayLondhe14 — APPROVED** [review]

**Pull request merged**